recipes/DragAndDropReactHooks/src/Main.purs

```diff
@@ -0,0 +1,114 @@
+module DragAndDropReactHooks.Main where
+
+import Prelude
+import Data.Foldable as Foldable
+import Data.Maybe (Maybe(..))
+import Data.Nullable as Nullable
+import Effect (Effect)
+import Effect.Exception as Exception
+import React.Basic.DOM as R
+import React.Basic.DOM.Events as DOM.Events
+import React.Basic.Events (SyntheticEvent)
+import React.Basic.Events as Events
+import React.Basic.Hooks (Component, (/\))
+import React.Basic.Hooks as React
+import Unsafe.Coerce as Coerce
+import Web.File.File as File
+import Web.File.FileList as FileList
+import Web.HTML as HTML
+import Web.HTML.Event.DataTransfer as DataTransfer
+import Web.HTML.Event.DragEvent (DragEvent)
+import Web.HTML.Event.DragEvent as DragEvent
+import Web.HTML.HTMLDocument as HTMLDocument
+import Web.HTML.HTMLElement as HTMLElement
+import Web.HTML.HTMLInputElement as HTMLInputElement
+import Web.HTML.Window as Window
+
+main :: Effect Unit
+main = do
+  maybeBody <- HTMLDocument.body =<< Window.document =<< HTML.window
+  case maybeBody of
+    Nothing -> Exception.throw "Could not find body."
+    Just body -> do
+      app <- mkApp
+      R.render (app unit) (HTMLElement.toElement body)
+
+mkApp :: Component Unit
+mkApp = do
+  window <- HTML.window
```

Comment:
Oops, I didn't end up needing a reference to the `window` I don't think, will remove